### ISSUE TITLE
slanted window status style with right aligned number leading to gap

### ISSUE THREAD

**Issue by NielsCloer** [catppuccin/tmux#389]
### Is there an existing issue outlining your problem?

- [X] I have searched the existing issues and they do not solve my problem.

### Describe your problem.

Problem as described in title.
I have set window number position to "right" and window status style to "slanted". This ends up leaving a gap to the left of the number.
As far as I have been able to identify this isn't what is supposed to happen.

### Paste your configuration.

```
unbind r
bind r source-file ~/.tmux.conf

set -g mouse on

bind-key h select-pane -L
bind-key j select-pane -D
bind-key k select-pane -U
bind-key l select-pane -R
bind-key x kill-pane
set -g base-index 1

set-option -g status-position top

set -g default-terminal "tmux-256color"

# Configure the catppuccin plugin
set -g @catppuccin_flavor "macchiato"

set -g @catppuccin_window_number_position "right"
set -g @catppuccin_window_status_style "slanted"

set -g @catppuccin_window_default_fill "number"
set -g @catppuccin_window_default_text "#W "

set -g @catppuccin_window_current_fill "number"
set -g @catppuccin_window_current_text "#{pane_current_path}"

set -g @catppuccin_status_left_separator  ""
set -g @catppuccin_status_right_separator " "
set -g @catppuccin_status_fill "icon"
set -g @catppuccin_status_connect_separator "no"

run ~/.config/tmux/plugins/catppuccin/tmux/catppuccin.tmux


set -g status-left ""
set -g status-right "#{E:@catppuccin_status_user}"
set -ag status-right "#{E:@catppuccin_status_directory}"

# List of plugins
set -g @plugin 'tmux-plugins/tpm'


# Other examples:
# set -g @plugin 'github_username/plugin_name'
# set -g @plugin 'github_username/plugin_name#branch'
# set -g @plugin 'git@github.com:user/plugin'
# set -g @plugin 'git@bitbucket.com:user/plugin'

# Initialize TMUX plugin manager (keep this line at the very bottom of tmux.conf)
run '~/.tmux/plugins/tpm/tpm'
```

### Attach screenshots.

![image](https://github.com/user-attachments/assets/8109fc06-5253-4561-a25e-3121ac2155ef)

### What tmux version are you seeing the issue on?

tmux 3.2a

**Comment by kjnsn** [comment]
This is a known issue that will be fixed in version 2. 

Also out of curiosity, why are you using tmux if your terminal supports tabs?

**Comment by NielsCloer** [comment]
> This is a known issue that will be fixed in version 2.
> 
> Also out of curiosity, why are you using tmux if your terminal supports tabs?

I have Ubuntu running on wsl as a sort of playground to experiment on. Tmux seemed like something good to have tried configuring for myself and having shortcuts for splits and windows within the linux VM instead of on windows just feels a little better for trying to get used to/learning linux.

**Comment by kjnsn** [comment]
Can you please try again with [79284da](https://github.com/catppuccin/tmux/commit/79284da665bf5d39d304e23df4165c8ac37f9b7a)?

**Comment by NielsCloer** [comment]
Still seeing the same issue although I am not 100% sure I'll have done everything correctly to switch to that commit since I feel like I remember seeing something about adding the version of catppuccin to config. (although that mightve been for tpm, which I am no longer using for catppuccin) 

Here's what I have done:
I have switched to your commit in the repository contained in ~/.tmux/plugins/tmux
uncommented the line in my config moving numbers to the right
restarted tmux using `tmux kill-server && tmux`

**Comment by kjnsn** [comment]
It seems like your changes haven't applied if you are still seeing the issue. If the number is still on the right, but you have commented out the line that sets numbers to be on the right, the config isn't getting picked up

**Comment by kjnsn** [comment]
Yeah you're definitely not actually running the code at that commit.

**Comment by NielsCloer** [comment]
Ok yeah that was pretty dumb. Was in the wrong directory but this doesn't exactly seem much better now.

![image](https://github.com/user-attachments/assets/62c1e45b-57a2-4c62-b799-64fa5c66c818)

Is this still a bug or is this now a mistake in my config?

**Comment by NielsCloer** [comment]
After some experimenting it seems to be caused by `set -ogq @catppuccin_status_background "none"`
Still unsure if this is partially my configs fault or if this is a bug
Here's the updated config:
![image](https://github.com/user-attachments/assets/b249ceae-4747-4d61-a577-f6590a2d9ac7)

And working as intended when background isn't disabled:
![image](https://github.com/user-attachments/assets/1f550dd7-a58b-4f83-836f-73163f3d839c)

**Comment by kjnsn** [comment]
Ah that makes a lot of sense then. 

You can do it yourself by instructing tmux to not draw a background on the separator cells.  I'm not sure though if ncurses is actually capable of doing that though

**Comment by davidfmatheson** [comment]
Looks fixed to me in latest:

<img width="521" alt="image" src="https://github.com/user-attachments/assets/78e9f1b0-2b48-4aec-b6b9-3d453ed94ba7">
